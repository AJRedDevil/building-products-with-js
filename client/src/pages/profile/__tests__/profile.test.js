// npm packages
import React from 'react';
import {MemoryRouter} from 'react-router';
import configureMockStore from 'redux-mock-store';

// our packages
import ProfilePage, {Profile} from '../index';

// create mockstore
const mockStore = configureMockStore();

const user = {
  id: 0,
  login: 'test',
};
const match = {
  isExact: true,
  params: {
    id: 0,
    login: 'test',
  },
  path: '/',
  url: 'http://localhost',
};

test('# Profile page wrapper', () => {
  const store = mockStore({auth: {user}, users: {user}});
  const wrapper = shallow(<ProfilePage match={match} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Profie page', () => {
  const getUser = (u) => {
    expect(u).toEqual(user);
  };

  const component = (
    <Profile
      getUser={getUser}
      match={match}
      user={user}
      loadedUser={user}
    />
  );
  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
});
