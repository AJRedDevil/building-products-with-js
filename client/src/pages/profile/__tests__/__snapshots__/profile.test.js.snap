// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Profie page 1`] = `
<div>
  <Navbar
    current="/profile/0"
    user={
      Object {
        "id": 0,
        "login": "test",
      }
    }
  />
  <Connect(User)
    edit={true}
    user={
      Object {
        "id": 0,
        "login": "test",
      }
    }
  />
</div>
`;

exports[`# Profile page wrapper 1`] = `
<Profile
  getUser={[Function]}
  loadedUser={
    Object {
      "id": 0,
      "login": "test",
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {
        "id": 0,
        "login": "test",
      },
      "path": "/",
      "url": "http://localhost",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "id": 0,
      "login": "test",
    }
  }
/>
`;
